build_tools/packaging/python/generate_release_index.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+"""Creates an HTML page for releases for `pip install --find-links` from a subdirectory in an S3 bucket.
+
+Sample usage:
+
+    ```bash
+    ./build_tools/packaging/python/generate_release_index.py \
+        --bucket=therock-dev-python \
+        --endpoint=s3.us-east-2.amazonaws.com \
+        --subdir=gfx110X-dgpu \
+        --output=index.html
+    ```
+
+S3 buckets in `us-east-2` currently used are:
+
+    * `therock-dev-python`
+    * `therock-nightly-python`
+"""
+
+import argparse
+import boto3
+import html
+import io
+import sys
+import textwrap
+
+
+def parse_arguments():
+    p = argparse.ArgumentParser()
+    p.add_argument(
+        "--bucket",
+        required=True,
+        help="S3 bucket which in which the subdirectory is indexed",
+    )
+    p.add_argument(
+        "--endpoint",
+        default="s3.us-east-2.amazonaws.com",
+        help="S3 endpoint used to form the URL",
+    )
+    p.add_argument(
+        "--subdir",
+        "--subdirectory",
+        required=True,
+        help="Subdirectory in S3 bucket which is indexed",
+    )
+    p.add_argument(
+        "--output",
+        default="-",
+        help="The file to write the HTML to or '-' for stdout (the default)",
+    )
+    return p.parse_args()
+
+
+def get_objects(bucket_name: str, subdir: str):
+    s3 = boto3.resource("s3")
+    bucket = s3.Bucket(bucket_name)
+
+    all_objects = bucket.objects.filter(Prefix=subdir)
+
+    objects = [obj.key.split(subdir + "/")[1] for obj in all_objects]
+    if "index.html" in objects:
+        objects.remove("index.html")
+
+    return objects
+
+
+def add_releases(objects: list, base_url: str, subdir: str, file: io.TextIOWrapper):
+
+    file.write(
+        f'    <h2> <a href="https://github.com/ROCm/TheRock">ROCm/TheRock</a> {subdir} packages</h2>\n'
+    )
```

Comment:
I guess we could also put "dev" or "nightly" in the header text, and maybe have them link to each other, e.g. "if you want the others, go here instead". WDYT?

Reply:
So far we don't have a complete set of Python `dev` packages. Let's add links once we have a set :)